### DISCUSSION TITLE
Is there any alternatives like `docker buildx imagetools` ?

### DISCUSSION THREAD

**Discussion by abcfy2** [containers/podman#20972] [General]
Hi there. Docker cli has a useful `docker buildx imagetools` could show remote image informations, like supported platforms and sha256.

```sh
# docker buildx imagetools inspect python
Name:      docker.io/library/python:latest
MediaType: application/vnd.docker.distribution.manifest.list.v2+json
Digest:    sha256:6d7fa2d5653e1d0eb464a672ded01f973e49e4a7ded59703c7bdcf6b92eac736
           
Manifests: 
  Name:      docker.io/library/python:latest@sha256:c50322f5f9b2a3b7ac68dc3cf03e5b29d7f51faa58b8321d975f028eb0c00a73
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/amd64
             
  Name:      docker.io/library/python:latest@sha256:037769815b6dd0308f753f767a8c342bc19352e11dc90ce9d626de91418a3484
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v5
             
  Name:      docker.io/library/python:latest@sha256:470bd7fc3c9a7c210117b0817d43b99edd20b7f00d3e786f9f3c15400646957f
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm/v7
             
  Name:      docker.io/library/python:latest@sha256:988ea4571fd487040a1da82cb35e043de818440a311ddc90f2b9f8dbac54895c
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/arm64/v8
             
  Name:      docker.io/library/python:latest@sha256:5640f12b97e7adb328d974a8e7f7f646d0f1ae8c7e2dd37a870c5b66672c744c
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/386
             
  Name:      docker.io/library/python:latest@sha256:ae51250796ac9e55dab64c7b9dbae8294059ce4eb043e7695efb3f1f423f5253
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/ppc64le
             
  Name:      docker.io/library/python:latest@sha256:bab5712f7ee4c9ea067db88c9b0314b7e808b2e33d078a284cef9e2bec78cbc0
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  linux/s390x
             
  Name:      docker.io/library/python:latest@sha256:33684f884e422f87e4a47a182d96b4d7c15b93781ce1627ef067dd46b980ad5e
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  windows/amd64
  OSVersion: 10.0.20348.2113
             
  Name:      docker.io/library/python:latest@sha256:a13d5420b7c208547f949dcb6c46353020421859c76066b562f35ab1ecdf2a5e
  MediaType: application/vnd.docker.distribution.manifest.v2+json
  Platform:  windows/amd64
  OSVersion: 10.0.17763.5122
```

Is there any alternatives in podman ? 

Thanks.

**Comment by rhatdan** [comment]
This feels like skopeo functionality.
Thoughts @mtrmac @vrothberg

**Reply by vrothberg** [reply]
Yes, this looks similar to `skopeo inspect --raw docker://python|jq .`.

As for the entirety of `buildx imagetools`, I have no idea.

**Reply by rhatdan** [reply]
Bottom line is we have access to the data, it is about someone stepping up and implementing it.

**Reply by vrothberg** [reply]
Why would we implement it?  skopeo already shows the data.

I personally don't think it's sustainable to chase everything buildx does.  In this case, I think that buildx wanted something similar to `skopeo inspect` actually.

**Reply by rhatdan** [reply]
Sure, although forcing everyone to use multiple tools is also a bit painful.

**Reply by mtrmac** [reply]
Skopeo is tracking https://github.com/containers/skopeo/issues/1283 . That’s not just what the example above shows, the output format needs to handle single-platform images, artifacts, or any future kinds of things that can be tagged on a registry.

---

> I personally don't think it's sustainable to chase everything buildx does.

That’s… a just an aspect of the general question about whether Podman wants to chase everything /usr/bin/docker decides to add.

**Reply by rhatdan** [reply]
As long as we are known for `alias docker=podman` we have to consider requests like this.  Buildx is a step beyond, and so far we have just been handling the `docker buildx build` path, which seems to be what most Docker users use.

**Reply by abcfy2** [reply]
Any update ? Do we have plans to implement this feature or not?

**Reply by rhatdan** [reply]
If someone steps up from the community then we would definitely consider it.  For now the core team is not working on this feature, as we have many other priorities.
